src/System.Management.Automation/help/BaseCommandHelpInfo.cs

```diff
@@ -393,8 +393,20 @@ internal override PSObject[] GetParameter(string pattern)
                 return base.GetParameter(pattern);
             }
 
+            // The Maml format simplifies array fields containing only one object
+            // by transforming them into the objects themselves. To ensure the consistency
+            // of the help command result we change it back into an array.
+            var param = prmts.Properties["parameter"].Value;
+            PSObject paramPSObj = param as PSObject;
+            PSObject[] paramAsPSObjArray = new PSObject[1];
+
+            if (paramPSObj != null)
+            {
+                paramAsPSObjArray[0] = paramPSObj;
+            }
+
             PSObject[] prmtArray = (PSObject[])LanguagePrimitives.ConvertTo(
-                prmts.Properties["parameter"].Value,
+                paramAsPSObjArray != null ? paramAsPSObjArray : param,
```

Comment:
You could probably condense much of this into the ternary statement, if I'm reading this right...

```suggestion
                paramPSObjArray == null ? paramPSObjArray : new[] { (PSObject)prmts.Properties["parameter"].Value },
```

Reply:
You can do `PSObject.AsPSObject(obj)` to safely convert object -> PSObject, but I'm not sure whether that would completely resolve the issue.

Reply:
I'm not 100% sure, but I think that's more or less just what ConvertTo would do itself... I'd have to chase down the convoluted code paths in there, though. I'm sure it's OK to use ConvertTo, though. 🙂